src/js/view/components/nextuptooltip.js

```diff
@@ -26,35 +25,34 @@ define([
             this.closeButton.setAttribute('aria-label', this.nextUpClose);
             this.tooltip = this.content.querySelector('.jw-nextup-tooltip');
 
+            var model = this._model;
             // Next Up is hidden until we get a valid NextUp item from the nextUp event
-            this.showNextUp = false;
-            this.streamType = undefined;
+            model.set('nextUpEnabled', false);
 
             // Events
-            this._model.on('change:mediaModel', this.onMediaModel, this);
-            this._model.on('change:streamType', this.onStreamType, this);
-            this._model.on('change:nextUp', this.onNextUp, this);
+            model.on('change:mediaModel', this.onMediaModel, this);
+            model.on('change:streamType', this.onStreamType, this);
+            model.on('change:nextUp', this.onNextUp, this);
+            model.on('change:nextUpVisible', this.toggle, this);
+            model.on('change:nextUpSticky', this.toggle, this);
 
             // Listen for duration changes to determine the offset from the end for when next up should be shown
-            this._model.on('change:duration', this.onDuration, this);
+            model.on('change:duration', this.onDuration, this);
             // Listen for position changes so we can show the tooltip when the offset has been crossed
-            this._model.on('change:position', this.onElapsed, this);
+            model.on('change:position', this.onElapsed, this);
 
-            this.onMediaModel(this._model, this._model.get('mediaModel'));
+            this.onMediaModel(model, model.get('mediaModel'));
 
             // Close button
             new UI(this.closeButton, {'directSelect': true})
-                .on('click tap', this.hide, this);
+                .on('click tap', function() {
+                    model.set('nextUpSticky', false);
+                    model.set('nextUpVisible', false);
+                    this.toggle(model, false);
```

Comment:
If the change handler for `nextUpSticky` & `nextUpVisible` is `toggle`, do you need to call toggle after setting either of these to false?